pkg/api/rpms.go

```diff
@@ -26,6 +26,31 @@ type RepositoryRpmCollectionResponse struct {
 	Links Links            `json:"links"` // Links to other pages of results
 }
 
+type SnapshotErrataListRequest struct {
+	UUID     string   `param:"uuid"`     // Identifier of the repository
+	Search   string   `query:"search"`   // Errata Id to optionally filter-on
+	Type     []string `query:"type"`     // Type to optionally filter-on
+	Severity []string `query:"severity"` // Severity to optionally filter-on
+}
+
+type SnapshotErrata struct {
+	Id              string  `json:"id"`
+	ErrataId        string  `json:"errata_id"`              // ID of the errata
+	Title           string  `json:"title"`                  // Title of the errata
+	Summary         string  `json:"summary"`                // Summary of the errata
+	Description     string  `json:"description"`            // Description of the errata
+	IssuedDate      string  `json:"issued_date"`            // IssuedDate of the errata
+	UpdateDate      *string `json:"updated_date,omitempty"` // UpdateDate of the errata
```

Comment:
For the null type issue, I think it would be better to change the type of `UpdateDate` to `string`, rather than use `omitempty`.

Reply:
We generally don't use pointers and omit fields in response structs, so I think it would be more consistent with other endpoints that way.

Reply:
hmm that's what i had initially, though i did update the UpdateDate in tang to be a pointer as well to get past this: https://github.com/content-services/tang/pull/6#issuecomment-2061391990

Reply:
It makes sense to change that in Tang because the type in the pulp database allowed null values. Here, you should be able to just check if it's nil in the struct from Tang, and then make it empty string if it is.

Reply:
To elaborate, I think the types returned by Tang should be as close to types as they are defined in the pulp database. But in practice, we're going interpret empty string and null the same way, so I think it makes sense to remove that nuance here.

Reply:
ah gotcha, that makes sense :)